apps/api/v2/src/ee/bookings/2024-08-13/services/booking-location.service.ts

```diff
@@ -0,0 +1,91 @@
+import { BookingsRepository_2024_08_13 } from "@/ee/bookings/2024-08-13/repositories/bookings.repository";
+import { BookingsService_2024_08_13 } from "@/ee/bookings/2024-08-13/services/bookings.service";
+import { InputBookingsService_2024_08_13 } from "@/ee/bookings/2024-08-13/services/input.service";
+import { ApiAuthGuardUser } from "@/modules/auth/strategies/api-auth/api-auth.strategy";
+import { UsersRepository } from "@/modules/users/users.repository";
+import { Injectable, Logger, NotFoundException } from "@nestjs/common";
+
+import type {
+  UpdateBookingLocationInput_2024_08_13,
+  BookingInputLocation_2024_08_13,
+} from "@calcom/platform-types";
+import { Booking } from "@calcom/prisma/client";
+
+@Injectable()
+export class BookingLocationService_2024_08_13 {
+  private readonly logger = new Logger("BookingLocationService_2024_08_13");
+
+  constructor(
+    private readonly bookingsRepository: BookingsRepository_2024_08_13,
+    private readonly bookingsService: BookingsService_2024_08_13,
+    private readonly usersRepository: UsersRepository,
+    private readonly inputService: InputBookingsService_2024_08_13
+  ) {}
+
+  async updateBookingLocation(
+    bookingUid: string,
+    input: UpdateBookingLocationInput_2024_08_13,
+    user: ApiAuthGuardUser
+  ) {
+    const existingBooking = await this.bookingsRepository.getByUid(bookingUid);
+    if (!existingBooking) {
+      throw new NotFoundException(`Booking with uid=${bookingUid} not found`);
+    }
+    const { location } = input;
+
+    if (location) {
+      return await this.updateLocation(existingBooking, location, user);
+    }
+
+    return this.bookingsService.getBooking(existingBooking.uid, user);
+  }
+
+  getLocationValue(loc: BookingInputLocation_2024_08_13) {
```

Comment:
Can we move this function under `updateLocation` and make it private since it's not gonna be used outside of this service for now? And we move it under it so code reads top to bottom aka updateBookingLocation -> updateLocation -> getLocationValue and we can also make `updateLocation` private I think.